solvation_analysis/plotting.py

```diff
@@ -0,0 +1,294 @@
+    title : title of figure as a string
+    legend_label : title of legend as a string
+    x_axis : a string specifying "species" or "solution" to be graphed on the x_axis
+    series : Boolean (False for a bar graph; True for a line graph)
+
+    Returns
+    -------
+    fig : Plotly.Figure (generic plot)
+
+    """
+    # coerce solutions to a common name
+    for solution in coerce:
+        if solution in properties:
+            properties[solution][coerce[solution]] = properties[solution].pop(solution)
+
+    # filter out components of solution to only include those in keep_solvents
+    if keep_solvents:
+        for solution in properties:
+            try:
+                properties[solution] = {keep: properties[solution][keep] for keep in keep_solvents}
+            except KeyError:
+                # if argument for keep_solvents is invalid
+                raise Exception("Invalid value of keep_solvents. \n keep_solvents: " +
+                                str(keep_solvents) + "\n Valid options for keep_solvents: " +
+                                str(list(properties[solution].keys()))) from None
+
+    # generate figure and make a DataFrame of the data
+    fig = go.Figure()
+    df = pd.DataFrame(data=properties.values())
+    df.index = list(properties.keys())
+
+    if series and x_axis == "species":
+        # each solution is a line
+        df = df.transpose()
+        fig = px.line(df, x=df.index, y=df.columns, labels={"variable": legend_label})
+        fig.update_xaxes(type="category")
+    elif series and x_axis == "solution":
+        # each species is a line
+        fig = px.line(df, x=df.index, y=df.columns, labels={"variable": legend_label})
+        fig.update_xaxes(type="category")
+    elif not series and x_axis == "species":
+        # each solution is a bar
+        df = df.transpose()
+        fig = px.bar(df, x=df.index, y=df.columns, barmode="group", labels={"variable": legend_label})
+    elif not series and x_axis == "solution":
+        # each species is a bar
+        fig = px.bar(df, x=df.index, y=df.columns, barmode="group", labels={"variable": legend_label})
+
+    fig = format_graph(fig, title, x_label, y_label)
+    return fig
+
+
+def compare_free_solvents(solutions):
+    # this should be a grouped vertical bar chart or a line chart
+    # 1.0 should be marked and annotated with a dotted line
+    fig = compare_solvent_dicts()
+    return
+
+
+def compare_pairing(solutions, coerce=None, keep_solvents=None, x_label="Solvent", y_label="Pairing", title="Graph of Pairing Data", legend_label="Legend", **kwargs):
+    # this should be a grouped vertical bar chart or a line chart
+    # 1.0 should be marked and annotated with a dotted line
+    """
+    Compares the pairing of multiple solutions.
+    Parameters
+    ----------
+    solutions : a dictionary of Solution objects
+    coerce : a dictionary where the keys are strings of solvent names and the values are
```

Comment:
Let's think of a slightly more explicit name for coerce, maybe `coerce_solvent_names`